ctapipe/image/muon/muon_reco_functions.py

```diff
@@ -153,8 +153,10 @@ def analyze_muon_event(event, params=None, geom_dict=None):
         # diameter of 0.11, all cameras are perfectly circular   cam_rad =
         # np.sqrt(numpix*0.11/(2.*np.pi))
 
-        if(np.sum(pix_im > tailcuts[0]) > 0.1 * minpix
-           and np.sum(pix_im) > minpix
+        if(len(pix_im[pix_im > tailcuts[0]]) > 0.1 * minpix
+           # > 0.1 * minpix above threshold
+           and len(pix_im[pix_im > 0]) > minpix
+           # Npix composing the ring > minpix
```

Comment:
I think these comments show that this part of the code could profit from some documentation.
I personally like a refactoring process sometimes called: "breaking out functions" better than putting comments.

The reason is, that breaking out functions, often leads to code, which is better testable. 

In this case, instead of a comment, one could create a function e.g. similar to this one:
```python
def Npix_composing_ring(image):
    return np.sum(image > 0)
#and
def Npix_above_threshold(image, threshold=tailcuts[0]):
    return np.sum(image > threshold)

#then the if-clause would read:
if Npix_above_threshold(pix_im) > 0.1 * minpix and Npix_composing_ring(pix_im) > minpix:
    # do stuff
```

Comments tend to become "stale" and "smell ugly" after a time, especially when somebody changes the `0.1` in the code to `0.15` and forgets to update the comment.

Also with these new functions, you might want to go ahead and implement a little unit test, to make sure they really work as expected...

Reply:
Thanks for the review.
Ok, I'll implement look into the proposed changes.

Just added a test module for ring_completeness and filled it in muon_reco_functions, please do not merge yet

Reply:
I implemented the two proposed functions in `image/muon/features.py` (although I do not know if another location would be better for them...) and two test modules to test them.
I also modified the code in `muon_reco_functions.py` to call the functions and do not need to include comments.